AntlrKotlinTask is not compatible with Gradle 8.12

When using AntlrKotlinTask in a Gradle 8.12 project I run into the following exception:

```
Caused by: java.lang.NoSuchMethodError: 'org.gradle.process.internal.AbstractExecHandleBuilder org.gradle.process.internal.JavaExecHandleBuilder.redirectErrorStream()'
	at com.strumenta.antlrkotlin.gradle.internal.AntlrWorkerManager.createWorkerProcess(AntlrWorkerManager.kt:40)
	at com.strumenta.antlrkotlin.gradle.internal.AntlrWorkerManager.runWorker(AntlrWorkerManager.kt:17)
	at com.strumenta.antlrkotlin.gradle.AntlrKotlinTask.execute(AntlrKotlinTask.kt:145)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:125)
	at org.gradle.api.internal.project.taskfactory.IncrementalTaskAction.doExecute(IncrementalTaskAction.java:45)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:51)
	at org.gradle.api.internal.project.taskfactory.IncrementalTaskAction.execute(IncrementalTaskAction.java:26)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:29)
	at org.gradle.api.internal.tasks.execution.TaskExecution$3.run(TaskExecution.java:244)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:47)
	at org.gradle.api.internal.tasks.execution.TaskExecution.executeAction(TaskExecution.java:229)
	at org.gradle.api.internal.tasks.execution.TaskExecution.executeActions(TaskExecution.java:212)
	at org.gradle.api.internal.tasks.execution.TaskExecution.executeWithPreviousOutputFiles(TaskExecution.java:195)
	at org.gradle.api.internal.tasks.execution.TaskExecution.execute(TaskExecution.java:162)
	at org.gradle.internal.execution.steps.ExecuteStep.executeInternal(ExecuteStep.java:105)
	at org.gradle.internal.execution.steps.ExecuteStep.access$000(ExecuteStep.java:44)
	at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:59)
	at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:56)
```

--- a/antlr-kotlin-gradle-plugin/src/main/kotlin/com/strumenta/antlrkotlin/gradle/internal/AntlrWorkerManager.kt
+++ b/antlr-kotlin-gradle-plugin/src/main/kotlin/com/strumenta/antlrkotlin/gradle/internal/AntlrWorkerManager.kt
@@ -3,6 +3,7 @@
 package com.strumenta.antlrkotlin.gradle.internal
 
 import org.gradle.api.file.FileCollection
+import org.gradle.process.internal.JavaExecHandleBuilder
 import org.gradle.process.internal.worker.RequestHandler
 import org.gradle.process.internal.worker.WorkerProcessFactory
 import java.io.File
@@ -37,8 +38,19 @@ internal class AntlrWorkerManager {
     javaCommand.workingDir = workingDir
     javaCommand.maxHeapSize = spec.maxHeapSize
     javaCommand.systemProperty("ANTLR_DO_NOT_EXIT", "true")
-    javaCommand.redirectErrorStream()
+    javaCommand.redirectErrorStreamCompat()
 
     return builder.build()
+  }
+
+  /**
+   * Merge the process' error stream into its output stream.
+   *
+   * Solves a compatibility issue with Gradle 8.12+.
+   * See [antlr-kotlin/issues/201](https://github.com/Strumenta/antlr-kotlin/issues/201).
+   */
+  private fun JavaExecHandleBuilder.redirectErrorStreamCompat() {
+    val method = this::class.java.getMethod("redirectErrorStream")
+    method.invoke(this)
   }
 }

